lib/task/actor/actorMoveRelatedDescriptionsTask.class.php

```diff
@@ -0,0 +1,121 @@
+      new sfCommandOption('env', null, sfCommandOption::PARAMETER_REQUIRED, 'The environment', 'cli'),
+      new sfCommandOption('connection', null, sfCommandOption::PARAMETER_REQUIRED, 'The connection name', 'propel'),
+      new sfCommandOption('skip-index', null, sfCommandOption::PARAMETER_NONE, "Skip Elasticsearch indexing"),
+    ));
+
+    $this->namespace = 'actor';
+    $this->name = 'move-related-descriptions';
+    $this->briefDescription = 'Move descriptions from an actor to another';
+    $this->detailedDescription = <<<EOF
+Move related descriptions from a source actor to a target actor, including events and relations.
+EOF;
+  }
+
+  protected function execute($arguments = array(), $options = array())
+  {
+    parent::execute($arguments, $options);
+
+    if (null === $source = QubitActor::getBySlug($arguments['source']))
+    {
+      throw new sfException('An actor with slug "'.$arguments['source'].'" could not be found.');
+    }
+
+    if (null === $target = QubitActor::getBySlug($arguments['target']))
+    {
+      throw new sfException('An actor with slug "'.$arguments['target'].'" could not be found.');
+    }
+
+    $sourceName = $source->getAuthorizedFormOfName(array('cultureFallback' => true));
+    $targetName = $target->getAuthorizedFormOfName(array('cultureFallback' => true));
+    $this->log('Moving related descriptions from "'.$sourceName.'" to "'.$targetName.'" ...');
+
+    // Amalgamate related description ids before update
+    $relatedIoIds = array();
+    if (!$options['skip-index'])
+    {
+      $sql = "SELECT event.object_id FROM event
+        JOIN object ON event.object_id=object.id
+        WHERE event.actor_id=:sourceId
+        AND object.class_name='QubitInformationObject'
+        UNION ALL
+        SELECT relation.subject_id FROM relation
+        JOIN object ON relation.subject_id=object.id
+        WHERE relation.object_id=:sourceId
+        AND relation.type_id=:typeId
+        AND object.class_name='QubitInformationObject'";
+      $params = array(':sourceId' => $source->id, ':typeId' => QubitTerm::NAME_ACCESS_POINT_ID);
+      $relatedIoIds = QubitPdo::fetchAll($sql, $params, array('fetchMode' => PDO::FETCH_COLUMN));
+    }
+
+    // Move all events
+    $sql = "UPDATE event
+      JOIN object ON event.object_id=object.id
+      SET event.actor_id=:targetId
+      WHERE event.actor_id=:sourceId
+      AND object.class_name='QubitInformationObject'";
+    $params = array(':targetId' => $target->id, ':sourceId' => $source->id);
+    $updatedCount = QubitPdo::modify($sql, $params);
+
+    // Move name access point relations
+    $sql = "UPDATE relation
+      JOIN object ON relation.subject_id=object.id
+      SET relation.object_id=:targetId
+      WHERE relation.object_id=:sourceId
+      AND relation.type_id=:typeId
+      AND object.class_name='QubitInformationObject'";
+    $params = array(':targetId' => $target->id, ':sourceId' => $source->id, ':typeId' => QubitTerm::NAME_ACCESS_POINT_ID);
```

Comment:
This raises a question for me about line lengths and wrapping.  I've been trying to wrap my code lines according to the rules in https://www.php-fig.org/psr/psr-12/ - e.g. aiming for 80 chars per line or less, but allowing 120 chars for legibility, and avoiding >120 chars per line except in extreme cases.   I don't see any recommendations in The [AtoM Coding Standard](https://wiki.accesstomemory.org/wiki/Development/Coding_standard) or [Symfony Coding Standards](https://symfony.com/doc/current/contributing/code/standards.html) for line length, but do we have any consensus among the AtoM devs on line lengths?

Reply:
I like that approach @djjuhasz! And I'd love to see a CI process in AtoM that enforces those rules at some point ...

https://github.com/FriendsOfPHP/PHP-CS-Fixer
https://github.com/squizlabs/PHP_CodeSniffer

Reply:
@jraddaoui I would also like to add a linter or formatter to our development workflow at some point, but doing so will mean changing some of our current coding standards I think.  I'll make a task for myself to add the line length standard to our coding standard as a first step.

Reply:
I'm not sure how much can be configured with those tools, but we may be able to reflect our current coding standards in them. Although, I don't think it's a bad idea to use an opinionated code formatter, even if we have to change our style.